diff -up squashfs/squashfs-tools/Makefile.cflags squashfs/squashfs-tools/Makefile
--- squashfs/squashfs-tools/Makefile.cflags	2008-08-26 03:15:16.000000000 -0400
+++ squashfs/squashfs-tools/Makefile	2008-12-18 06:59:01.000000000 -0500
@@ -2,7 +2,7 @@ INSTALL_DIR = /usr/local/bin
 
 INCLUDEDIR = .
 
-CFLAGS := -I$(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE -O2
+CFLAGS := $(RPM_OPT_FLAGS) -I$(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE -O2
 
 all: mksquashfs
 
